core/controllers/editor.py

```diff
@@ -190,30 +190,14 @@ def get(self, exploration_id: str) -> None:
         # 'apply_draft' and 'v'(version) are optional parameters because the
         # exploration history tab also uses this handler, and these parameters
         # are not used by that tab.
-        assert self.user_id is not None
         assert self.normalized_request is not None
         version = self.normalized_request.get('v')
         apply_draft = self.normalized_request['apply_draft']
 
-        user_settings = user_services.get_user_settings(
-            self.user_id, strict=False
-        ) if self.user_id else None
-        has_seen_editor_tutorial = False
-        has_seen_translation_tutorial = False
-        if user_settings is not None:
-            if user_settings.last_started_state_editor_tutorial:
-                has_seen_editor_tutorial = True
-            if user_settings.last_started_state_translation_tutorial:
-                has_seen_translation_tutorial = True
-
         try:
             exploration_data = exp_services.get_user_exploration_data(
```

Comment:
I think a method called get_user_exploration_data should return a UserExplorationData object.

Do we need two separate objects -- UserExplorationData and AugmentedUserExplorationData? Or, if this is not specific to the exploration + user pair, should we pass the other fields separately -- basically formatting the rendered_json as a dict with keys 'user_exploration_data', 'show_state_editor_tutorial_on_load', 'show_state_translation_tutorial_on_load', 'exploration_is_linked_to_story'?

Reply:
Sorry for the delay, and I agree we should change the approach a bit and return a UserExplorationData object. we don't need two objects though as it's a bit unnecessary.

Instead of updating the json on the go we should stick with the domain object but change it to UserExplorationData as we aren't really adding new things to it.

The only properties that are user specific are 'show_state_editor_tutorial_on_load' 'show_state_translation_tutorial_on_load' and 'email_preferences' which we should set to default values in the domain object and only update the required properties when needed.

And we should avoid setting 'show_state_translation_tutorial_on_load' and 'show_state_editor_tutorial_on_load' in the `get_user_exploration_data` function, as these two properties are used in fetching exploration data only and not while updating exploration/draft.

Reply:
I guess something along those lines seems reasonable, but just to clarify one thing conceptually: "user exploration data" refers specifically to info that is tied to a (user, exploration) pair. So anything that relates only to the exploration, or only to the user, isn't really considered "user exploration data".

At least, this is the current conceptual convention used in the codebase, so let's stick to that in any modifications that are made.

Reply:
Understood, thanks!